### ISSUE TITLE
Editing a field creates a new object with Grammarly

### ISSUE THREAD

**Issue by HexM7** [Glitchii/embedbuilder#6]
When editing an existing field, a new value object is created. A simple fix to this is to make minor changes to the **index.js** file. At line `454`, replace 
```javascript
let jsonField = json.embed.fields[Array.from(fields.children).indexOf(field)];
```
with
```javascript
let jsonField = json.embed.fields[Array.from(fields.children).indexOf(field)-2];
```

**Source:** Invalid array index

**Comment by Glitchii** [comment]
I cannot seem to replicate this problem. Works fine when I edit an existing field. In fact adding a `-2` just didn't work properly and the first two fields where being ignored since you are remove 2 from the index.

Which is why this issue is confusing... What are the steps to get to this problem?

**Comment by HexM7** [comment]
**Steps to reproduce:**
- Visit https://glitchii.github.io/embedbuilder/
- Erase everything
- Add a new field
- Try to edit the field value/name

I observed that the changes are not reflected on the right side (embed), and when I checked the JSON code, there was an extra `value` object outside the filed object:
![Screenshot (173)](https://user-images.githubusercontent.com/77036902/124470224-fb040e00-ddb8-11eb-9dd9-1c0e93506d71.png)

The issues seem to be arising from the line `454` in **index.js** due to inaccurate array index (I may be wrong) as the `jsonField` variable returns undefined resulting in a new object.

`-2` was really a bad idea :p

**Comment by Glitchii** [comment]
I still can't seem to replicate the it. I followed the exact steps you gave.

https://user-images.githubusercontent.com/46096865/124517478-eb74cd00-dd98-11eb-8621-7e21e6737db4.mp4

**Comment by Glitchii** [comment]
Still can't replicate the problem. I followed your exact steps

**Comment by Glitchii** [comment]
Looking at your screenshot, I think I get what you mean now. The last fields were not displayed because you did not have values. If you were to send this embed to Discord, it would not accept it since there has to be a value. The embed on the right does not display the field assuming you are still typing and you will add the field name, then it will display it when it's in the right format

**Comment by HexM7** [comment]
It still does not work with field names. Kindly see the attached video:

https://user-images.githubusercontent.com/77036902/124544833-86c37c00-de45-11eb-8f58-fa12a1f8ca66.mp4

**Comment by Glitchii** [comment]
Turns out this issue is caused by the Grammarly extension. Looks like the extension changes the HTML around and messes up the texarea, or probably re-writes it, which I suppose removes the input event listener on the textarea. And that's why this problem only happens in the textarea box and not on the input box. I will re-open this issue and rename it to "Editing a field creates a new object with Grammarly" while I dig some more on the way Grammarly works and probably come to a fix. Thanks.

**Comment by HexM7** [comment]
This fixed it! It was the extension that was messing up with the textarea. Thanks!

**Comment by Glitchii** [comment]
I have just made a new commit which allows it to work with the extension.
Grammarly adds a new element named `grammarly-extension`, All I had to do is to filter out only the div elements by replacing this line.
```js
let jsonField = json.embed.fields[Array.from(fields.children).indexOf(field)];
```
with this.
```js
let jsonField = json.embed.fields[Array.from(fields.children).filter(e => e.className === 'field').indexOf(field)];
```
This should now be fixed after the commit 33a798cc44062611ddf1d7f6cf2ee0cc3e3cb68d, please let me know if it's working fine now on your end.
